testcafe/tests/pages/drive-model.js

```diff
@@ -85,4 +165,195 @@ export default class Page {
       )
       .ok('Drive is not current app')
   }
+
+  async openActionMenu() {
+    isExistingAndVisibile(this.toolbar_files, 'toolbar_files')
+    isExistingAndVisibile(this.btnMoreMenu, `[...] button`)
+    await t.click(this.btnMoreMenu)
+    isExistingAndVisibile(this.coz_menu_inner, 'Cozy inner menu')
+  }
+  //@param {String} newFolderName
+  async addNewFolder(newFolderName) {
+    const breadcrumbStart = await this.getbreadcrumb()
+    const rowCountStart = await this.getContentRowCount('Before')
+
+    await this.openActionMenu()
+    isExistingAndVisibile(this.btnAddFolder, 'Add Folder button')
+    await t.click(this.btnAddFolder)
+
+    const rowCountEnd = await this.getContentRowCount('After')
+    await t.expect(rowCountEnd).eql(rowCountStart + 1) //New content line appears
+
+    isExistingAndVisibile(this.foldersNamesInputs, 'Folder Name input')
+    await t
+      .typeText(this.foldersNamesInputs, newFolderName)
+      .pressKey('enter')
+      .expect(this.foldersNamesInputs.exists)
+      .notOk('Edition mode still on') //No folder in edition mode -> No input on page
+      .expect(this.foldersNames.withText(newFolderName).exists)
+      .ok(`No folder named ${newFolderName}`)
+
+    const breadcrumbEnd = await this.getbreadcrumb()
+    await t.expect(breadcrumbEnd).eql(breadcrumbStart)
+  }
+
+  //@param {String} folderName
+  async goToFolder(folderName) {
+    const breadcrumbStart = await this.getbreadcrumb()
+    await t
+      .expect(this.foldersNames.withText(folderName).exists)
+      .ok(`No folder named ${folderName}`)
+      .click(this.foldersNames.withText(folderName))
+
+    await this.waitForLoading()
+
+    const breadcrumbEnd = await this.getbreadcrumb()
+    await t.expect(breadcrumbEnd).eql(`${breadcrumbStart} > ${folderName}`)
+
+    console.log(`Navigation into ${breadcrumbEnd} OK!`)
+  }
+
+  //@param {String Array} files: path to files to upload.
+  async uploadFiles(files) {
+    const numOfFiles = files.length
+    const rowCountStart = await this.getContentRowCount('Before')
+
+    console.log('Uploading ' + numOfFiles + ' file(s)')
+
+    isExistingAndVisibile(this.btnUpload)
+    await t.setFilesToUpload(this.btnUpload, files)
+
+    isExistingAndVisibile(this.divUpload, 'Upload pop-in')
+    isExistingAndVisibile(this.divUploadSuccess, 'successfull Upload pop-in')
+    isExistingAndVisibile(this.alertWrapper, '"successfull" modal alert')
+    await t
+      .expect(this.divUpload.child('h4').innerText)
+      .match(
+        new RegExp('([' + numOfFiles + '].*){2}'),
+        'Numbers of pictures uploaded does not match'
+      )
+    await t.takeScreenshot()
+    const rowCountEnd = await this.getContentRowCount('After')
+    await t.expect(rowCountEnd).eql(rowCountStart + 1) //New content line appears
+  }
+
+  //NOT FINISH !!!
```

Comment:
And uh… can we finish it before merging 😁 ?
If not, I suggest we remove the code until it's working as we want :)

Reply:
It's actually finish, but I forgot to remove the comment! ;)